graviti/manager/dataset.py

```diff
@@ -26,6 +26,36 @@
 LazyLists = NestedDict[str, LazyList[Any]]
 
 
+class DatasetAccessInfo(Mapping[str, str]):
+    """This class defines the basic structure of the dataset access info.
+
+    Arguments:
+        access_key: User's access key.
+        url: The URL of the graviti website.
+        owner: The owner of the dataset.
+        dataset: The name of the dataset, unique for a user.
+
+    """
+
+    def __init__(self, access_key: str, url: str, owner: str, dataset: str) -> None:
+        self.access_key = access_key
+        self.url = url
+        self.owner = owner
+        self.dataset = dataset
+
+    def __len__(self) -> int:
+        return 4
+
+    def __getitem__(self, key: str) -> str:
+        try:
+            return getattr(self, key)  # type: ignore[no-any-return]
+        except AttributeError:
+            raise IndexError(f"'{key}'") from None
+
+    def __iter__(self) -> Iterator[str]:
+        yield from ("access_key", "url", "owner", "dataset")
```

Comment:
do we need to set ReprMixin as the base of DatasetAccessInfo. then we can shall this tuple here and `_repr_attrs`.